Runtime/Scripts/PersistentConnection.cs

```diff
@@ -19,50 +38,122 @@ public class PersistentConnection : MonoBehaviour
     bool hasTCPPorts;
     int latencyUpdatesCounter;
     int locationUpdatesCounter;
+    Dictionary<ServerEventType, FindCloudletEventTrigger> event_trigger_dict;
+    #region MonoBehaviour Callbacks
+
+    private void Awake()
+    {
+      event_trigger_dict = new Dictionary<ServerEventType, FindCloudletEventTrigger>();
+      event_trigger_dict.Add(ServerEventType.EventAppinstHealth, FindCloudletEventTrigger.AppInstHealthChanged);
+      event_trigger_dict.Add(ServerEventType.EventCloudletState, FindCloudletEventTrigger.CloudletStateChanged);
+      event_trigger_dict.Add(ServerEventType.EventCloudletUpdate, FindCloudletEventTrigger.CloserCloudlet);
+      event_trigger_dict.Add(ServerEventType.EventCloudletMaintenance, FindCloudletEventTrigger.CloudletMaintenanceStateChanged);
+      event_trigger_dict.Add(ServerEventType.EventLatencyProcessed, FindCloudletEventTrigger.LatencyTooHigh);
+      event_trigger_dict.Add(ServerEventType.EventError, FindCloudletEventTrigger.Error);
+      //EVENT_INIT_CONNECTION and EVENT_UNKNOWN will be logged
+      //EVENT_LATENCY_REQUEST will be processed
+    }
 
     private void OnEnable()
     {
       startStreamingEvents += StartEdgeEvents;
       config = MobiledgeXIntegration.settings.edgeEventsConfig;
     }
 
-    void StartEdgeEvents(MobiledgeXIntegration mxi)
+    private void OnApplicationPause(bool pause)
     {
-      integration = mxi;
-      EdgeEventsConfig config = MobiledgeXIntegration.settings.edgeEventsConfig;
-      
-      if (config.newFindCloudletEventTriggers.Count == 0)
+      if (pause)
       {
-        FindCloudletEvent findCloudletEvent = new FindCloudletEvent()
+        StopAllCoroutines();// stop edge events streaming
+      }
+      else
+      {
+        // resume edge events streaming
+        if (locationUpdatesRunning)
+        {
+          StartCoroutine(OnIntervalEdgeEventsLocation(integration.matchingEngine.EdgeEventsConnection));
+        }
+        if (latencyUpdatesRunning)
         {
-          trigger = FindCloudletEventTrigger.Error
-        };
-        integration.NewFindCloudletHandler(EdgeEventsStatus.error, findCloudletEvent);
-        Debug.LogError(EdgeEventsError.MissingFindCloudletTrigger.ToString());
+          StartCoroutine(OnIntervalEdgeEventsLatency(integration.matchingEngine.EdgeEventsConnection, integration.GetHost()));
+        }
+      }
+    }
+
+    private void OnApplicationQuit()
+    {
+      StopAllCoroutines();// stop edge events streaming
+    }
+
+    private void OnDestroy()
+    {
+      StopAllCoroutines();// stop edge events streaming
+    }
+    #endregion
+
+    #region PersistentConnection Functions
+
+    bool ValidateConfigs()
+    {
+      if (config.newFindCloudletEventTriggers.Count == 1)//No FindCloudletTrigger except Error (Added By Default)
```

Comment:
Nit: Space, or rather, triggers.contains(Error)"

Reply:
Just an explanation, EventTrigger.Error is added by default, I am checking if there is any other triggers added.